src/main/java/de/hbt/pwr/profile/service/ProfileEntryService.java

```diff
@@ -69,12 +69,11 @@ public void deleteEntryWithId(Long id, Profile profile, NameEntityType nameEntit
     }
 
     public Skill updateProfileSkills(Skill skill, Profile profile) {
-        Optional<Skill> concurrent = profile.getSkills()
-                .stream().filter(s -> hasEqualName(s, skill)) // TODO
-                .findAny();
-        return concurrent
-                .map(s -> updateSkill(s, skill))
-                .orElseGet(() -> createNewInProfile(skill, profile));
+        Set<Skill> concurrent = profile.getSkills()
+                .stream().filter(s -> hasEqualName(s, skill))
+                .collect(Collectors.toSet());
+        concurrent.stream().forEach(s -> profile.getSkills().remove(s));
+        return createNewInProfile(skill, profile);
     }
```

Comment:
Optional<Skill> concurrent = profile.getSkills()
                .stream().filter(s -> hasEqualName(s, skill))
                .findAny();
        return concurrent
                .map(s -> updateSkill(s, skill))
                .orElseGet(() -> createNewInProfile(skill, profile));

Mit dieser Logik klappt es wieder. Ich verstehe den fix, aber er ist unnötig. Du hast in einem vorherigen commit die Logik ausgebaut gehabt, dass Dinge gemerged werden.
Es sollte nicht notwendig sein, hier mehrere concurrent skills zu finden, da es nur einen geben kann. Alles andere sollte rausgefiltert werden (schon vorher).
Was das hier macht:

- Duplikate Finden
- Alle Duplikat entfernen
- Skill neu anlegen

Das dauernde Skill neu anlegen sollte überflüssig sein.